modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -503,6 +504,10 @@ public void doDispose() {
   }
 
   protected ExecutionMediator createExecutionMediator() {
+    if (executionMediatorFactory != null) {
+      return executionMediatorFactory.get();
+    }
+
     return new DefaultExecutionMediator(extensionModel, componentModel, connectionManager, muleContext.getErrorTypeRepository());
```

Comment:
What if you always use a supplier, but when creating the instance, you create the default supplier?